persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,63 @@
+package com.baeldung.customer;
+
+import com.baeldung.config.PersistenceConfiguration;
+import com.baeldung.config.PersistenceProductConfiguration;
+import com.baeldung.config.PersistenceUserConfiguration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@DataJpaTest(excludeAutoConfiguration = { PersistenceConfiguration.class, PersistenceUserConfiguration.class, PersistenceProductConfiguration.class })
+@RunWith(SpringRunner.class)
+public class CustomerRepositoryIntegrationTest {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Autowired
+    private CustomerRepository repository;
+
+    @Before
+    public void before() {
+        entityManager.persist(new Customer("A", "A@example.com"));
+        entityManager.persist(new Customer("D", null));
+        entityManager.persist(new Customer("D", "D@example.com"));
+    }
+
+    @Test
+    public void givenQueryMethod_whenEmailIsNull_thenFoundByNullEmail() {
+        List<Customer> customers = repository.findByNameAndEmail("D", null);
+        assertEquals(1, customers.size());
```

Comment:
Let's add a blank line before the assertions begin. This separates the "when" and "then" parts of the test.